docs/book/src/getting-started.md

```diff
@@ -0,0 +1,540 @@
+Observe the validation schema within the CRD; this schema ensures that the Kubernetes API properly validates the Custom Resources (CRs) that are applied:
+
+From: `config/crd/bases/example.com.testproject.org_memcacheds.yaml`
+```yaml
+            description: MemcachedSpec defines the desired state of Memcached
+            properties:
+              containerPort:
+                description: Port defines the port that will be used to init the container
+                  with the image
+                format: int32
+                type: integer
+              size:
+                description: 'Size defines the number of Memcached instances The following
+                  markers will use OpenAPI v3 schema to validate the value More info:
+                  https://book.kubebuilder.io/reference/markers/crd-validation.html'
+                format: int32
+                maximum: 3 ## See here from the marker +kubebuilder:validation:Maximum=3
+                minimum: 1 ## See here from the marker +kubebuilder:validation:Minimum=1
+                type: integer
+            type: object
+
+```
+
+#### Sample of Custom Resources
+
+The manifests located under the "samples" directory serve as examples of Custom Resources that can be applied to the cluster.
+In this particular example, by applying the given resource to the cluster, we would generate a Deployment with a single instance size (see `size: 1`).
+
+From: `config/samples/example.com_v1alpha1_memcached.yaml`
+
+```shell
+apiVersion: example.com.testproject.org/v1alpha1
+kind: Memcached
+metadata:
+  name: memcached-sample
+spec:
+  # TODO(user): edit the following value to ensure the number
+  # of Pods/Instances your Operand must have on cluster
+  size: 1
+# TODO(user): edit the following value to ensure the container has the right port to be initialized
+  containerPort: 11211
+```
+
+### Reconciliation Process
+
+The reconciliation function plays a pivotal role in ensuring synchronization between resources and their specifications based on the business logic embedded within them. Essentially, it operates like a loop, continuously checking conditions and performing actions until all conditions align with its implementation. Here's a pseudo-code to illustrate this:
+
+```go
+reconcile App {
+
+// Check if a Deployment for the app exists, if not, create one
+// If there's an error, then restart from the beginning of the reconcile
+if err != nil {
+return reconcile.Result{}, err
+}
+
+// Check if a Service for the app exists, if not, create one 
+// If there's an error, then restart from the beginning of the reconcile
+if err != nil {
+return reconcile.Result{}, err
+}
+
+// Look for Database CR/CRD 
+// Check the Database Deployment's replicas size
+// If deployment.replicas size doesn't match cr.size, then update it
+// Then, restart from the beginning of the reconcile
```

Comment:
Can we add the comment that in this case we Requeue the reconcile request even when there is no error.